src/test/integration/integration-test.py

```diff
@@ -25,50 +25,64 @@ def test_get_authority_onids(self, endpoint='/spendingauthority'):
-        logging.debug(f'Request to made to {onid.lower()}')
-        logging.debug(f'Response for {actual_onid.lower()}')
-        self.assertEqual(actual_onid.lower(), onid.lower())
+            response_data = response.json()['data']
+            # Test case: GET /spendingauthority with valid authority onids
+            spending_schema = utils.get_resource_schema(
+                self, 'SpendingAuthorityResource'
+            )
+            utils.check_schema(self, response, spending_schema)
+            response_data = response.json()['data']
+            actual_onid = response_data['id']
+            logging.debug(f'Request to made to {onid}')
+            logging.debug(f'Response for {actual_onid}')
+            self.assertEqual(actual_onid.lower(), onid.lower())
+
+            self.checking_spending_limits(response_data['attributes'])
 
     # Test case: GET /spendingauthority spending limits
-    def checking_spending_limits(self, response):
+    def checking_spending_limits(self, attributes):
         # Checking that the lists of spending limits and indexes are unique
         # Checking valid onids with spending authority only
-        # To make sure that attributes is not empty
-        response_data = response.json()['data']
-        attributes = response_data['attributes']
-        self.assertTrue(attributes)
-        limits = attributes['limits']
-        # Checking that limits has at least one index
-        logging.debug(f'limits: {len(limits)}')
-        self.assertTrue(limits)
-        spendinglimit = []
+        # To make sure that attributes and index are not empty
+        non_empty_list_dict = {
+            'attributes': attributes,
+            'limits': attributes['limits']
+        }
+
+        for key, non_empty_list in non_empty_list_dict.items():
+            logging.debug(f'{key}: {len(non_empty_list)}')
+            self.assertTrue(non_empty_list)
+
+        spending_limit = []
         index_array = []
-        for limit in limits:
-            for index in limit['indexes']:
-                index_array.append(index)
-            spendinglimit.append(limit['spendingLimit'])
-            # Checking that each spending limit has at least one index
-            logging.debug(f'spending limit has: {limit["indexes"]}')
-            self.assertTrue(limit['indexes'])
+
+        for limit in non_empty_list_dict['limits']:
+            try:
+                for index in limit['indexes2']:
+                    index_array.append(index)
+                # Checking that each spending limit has at least one index
+                logging.debug(f'spending limit has: {limit["indexes"]}')
+                self.assertTrue(limit['indexes'])
+                try:
+                    spending_limit.append(limit['spendingLimit'])
+                except KeyError as error:
+                    logging.debug('spendingLimit does not exist')
+                    self.fail(error)
+            except KeyError as error:
```

Comment:
You have a nested try-catch block here and it's not necessary. Also, in my opinion, I think put the entire outer for-loop to the try-catch block make more sense.